demos/adjoint/forward.py

```diff
@@ -90,22 +140,24 @@
     transpose_nullspace=Z_nullspace,
     constant_jacobian=True,
 )
+# -
 
-# Create output file and select output_frequency
-output_file = VTKFile("vtu-files/output.pvd")
-dump_period = 10
+# Time Loop
+# ----------
+# We perform the time loop to solve the forward problem. During this loop, we solve the Stokes and energy systems, save the velocity field for later use, and periodically output the results.
 
-# Now perform the time loop:
-for timestep in range(0, max_timesteps):
+# +
+# Perform the time loop to solve the forward problem.
+for timestep in range(0, timesteps):
     stokes_solver.solve()
     energy_solver.solve()
 
-    # Storing velocity to be used in the objective F
-    checkpoint_file.save_function(u, name="Velocity", idx=timestep)
+    # Store the velocity field for use in the adjoint problem.
+    checkpoint_file.save_function(u, name="Velocity", idx=timestep, timestepping_info={"index": float(timestep), "delta_t": float(delta_t)})
```

Comment:
Do we have to convert the index to a float? That's weird!

Reply:
Yeah we do! Because firedrake enforces everything to be float for attributes.

Reply:
I agree it's a weird thing though! I think this was an oversight at the time by me, because Koki and I were just discussing having the time-step as timestepping history, not noting that this could be an integer.